code/datums/brain_damage/imaginary_friend.dm

```diff
@@ -136,6 +150,13 @@
 		client.images.Remove(human_image)
 	return ..()
 
+/mob/camera/imaginary_friend/proc/owner_speech(speaker, speech_args)
+	SIGNAL_HANDLER
+	var/list/listening = get_hearers_in_view(6, owner, SEE_INVISIBLE_MAXIMUM)
+	if(!(src in listening))
+		to_chat(src, "<span class='hear'>You hear a distant voice in your head...</span>")
+		to_chat(src, "<span class='game say'><span class='name'>[speaker]</span> <span class='message'>[say_quote(speech_args[SPEECH_MESSAGE])]</span></span>")
```

Comment:
Why are you doubling spans here?
Do classes game say exist?

Reply:
This is copied from existing say code, it's for chat tab filters.

Reply:
copy or not you can reduce the amount of spans by combining game say with name

Reply:
No, because name only wraps the name, the message won't filter properly otherwise